x/hard/keeper/withdraw.go

```diff
@@ -0,0 +1,76 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/kava-labs/kava/x/hard/types"
+)
+
+// Withdraw returns some or all of a deposit back to original depositor
+func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Coins) error {
+	// Get current stored LTV based on stored borrows/deposits
+	prevLtv, shouldRemoveIndex, err := k.GetStoreLTV(ctx, depositor)
```

Comment:
As a pure note, Ruaridh pointed out a cleaner way of doing the `Set/Update/Delete` pattern for ltv indexes in a CDP PR yesterday. I've created a card to update in hard as well.